handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1109,7 +1120,8 @@ public final synchronized void closeOutbound() {
 
     @Override
     public final synchronized boolean isOutboundDone() {
-        return isOutboundDone;
+        // check if there is anything left in the outbound buffer.
+        return outboundClosed && (isDestroyed() || SSL.pendingWrittenBytesInBIO(networkBIO) == 0);
```

Comment:
is checking `isDestroyed()` necessary? Setting the `destroyed` variable to "true" also means that `outboundClosed` will be set to `true`.